django/contrib/postgres/indexes.py

```diff
@@ -27,6 +27,13 @@ def deconstruct(self):
         kwargs['pages_per_range'] = self.pages_per_range
         return path, args, kwargs
 
+    def set_name_with_model(self, model):
+        try:
+            super().set_name_with_model(model)
+        except AssertionError:
```

Comment:
I think a more extensible solution would make `MAX_NAME_LENGTH` usage in `set_name_with_model()` a class attribute of the index which this subclass could override.